pkg/ipam/node.go

```diff
@@ -357,15 +357,21 @@ func (n *Node) UpdatedResource(resource *v2.CiliumNode) bool {
 	// dependent on caller not using the resource after this call.
 	resource = resource.DeepCopy()
 
-	n.ops.UpdatedNode(resource)
-
+	// Update n.resource before n.ops.UpdatedNode. This is to increase the chance
+	// perform a complete n.recalculate() when the nodeManager performs resync
+	// and the operator is startup.
```

Comment:
nit, better grammar:

```suggestion
	// of performing a complete n.recalculate() execution when the nodeManager performs a resync
	// and the operator is starting up.
```